File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        // key: item value, value: index
        // if key is contained in hashmap (dupe found) and their index difference <= k,
        // return true
        // else put item and new latest index into hashmap
        HashMap<Integer, Integer> hashMap = new HashMap<Integer, Integer>();
        for (int i = 0; i < nums.length; i++) {
            if (hashMap.containsKey(nums[i]) && Math.abs(i - hashMap.get(nums[i])) <= k) {
                return true;
            }
            hashMap.put(nums[i], hashMap.getOrDefault(nums[i], 0) + i);
        }
        return false;
    }
}

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hashMap = dict()
        for i,num in enumerate(nums):
            if num in hashMap:
                if i-hashMap[num] <= k:
                    return True
            hashMap[num] = i
        return False